test/src/system/block_formatting_test.coffee

```diff
@@ -444,6 +444,39 @@ testGroup "Block formatting", template: "editor_empty", ->
       assert.blockAttributes([5, 6], [])
       expectDocument("ab\n\ncd\n")
 
+  test "backspacing a newline in an empty block with adjacent formatted blocks", (expectDocument) ->
+    document = new Trix.Document [
+        new Trix.Block(Trix.Text.textForStringWithAttributes("abc"), ["heading1"])
+        new Trix.Block
+        new Trix.Block(Trix.Text.textForStringWithAttributes("d"), ["heading1"])
+      ]
+
+    replaceDocument(document)
+    getEditor().setSelectedRange(4)
+
+    pressKey "backspace", ->
+      document = getDocument()
+      assert.equal document.getBlockCount(), 2
+      assert.blockAttributes([0, 1], ["heading1"])
+      assert.blockAttributes([2, 3], ["heading1"])
+      expectDocument("abc\nd\n")
```

Comment:
This isn't technically testing *new* behavior. It is testing behavior that should work on master, but this PR broke until [this commit @ L143](https://github.com/basecamp/trix/pull/296/commits/8bc4d4ba22f4e766545d54ca9619e6d646a23f17).

Only discovered this was broken through manual testing, so I thought it could use a test. I suppose it doesn't need to be in this PR though?